apps/rehabstod/src/components/Table/TableLayout.tsx

```diff
@@ -0,0 +1,53 @@
+import { ReactNode } from 'react'
+import { FetchBaseQueryError } from '@reduxjs/toolkit/query'
+import { SerializedError } from '@reduxjs/toolkit'
+import { UnansweredCommunicationAlert } from '../error/ErrorAlert/UnansweredCommunicationAlert'
+import { TableContentAlert } from '../error/ErrorAlert/TableContentAlert'
+import { PrintButton } from '../PrintButton/PrintButton'
+import { useNavigateToStartPage } from '../../hooks/useNavigateToStartPage'
+
+export function TableLayout({
+  heading,
+  filters,
+  tableInfo,
+  modifyTableColumns,
+  tableContentError,
+  unansweredCommunicationError,
+  printable,
+  tableName,
+  children,
+}: {
+  heading: ReactNode
+  filters: ReactNode
+  tableInfo: ReactNode
+  modifyTableColumns: ReactNode
+  tableContentError?: (FetchBaseQueryError & { id?: string }) | (SerializedError & { id?: string })
+  unansweredCommunicationError?: boolean
+  tableName: string
+  printable: boolean
+  children: ReactNode
+}) {
+  useNavigateToStartPage()
+
+  return (
+    <div className="ids-content m-auto max-w-7xl py-10 px-2.5">
+      {heading}
+      {filters}
+      {tableContentError ? (
+        <TableContentAlert tableName={tableName} error={tableContentError} />
+      ) : (
```

Comment:
We can make the top part into a component, this is also used by the patient page, but the layout is a bit different.
```tsx
export function PageContainer({ children }: { children: ReactNode }) {
  return <div className="ids-content m-auto max-w-7xl py-10 px-2.5">{children}</div>
}
```

And we can move out heading, filters and children. `LUCertificate` would look something like:

```tsx
 <PageContainer>
  <TableHeadingForUnit ... />
  <LUCertificatesFilters ... />
  {error && <TableContentAlert ... /> }
  {luCertificatesInfo?.questionAndAnswersError && (
    <div className="pb-10">
      <UnansweredCommunicationAlert />
    </div>
  )}
  <TableLayout printable tableInfo={...} modifyTableColumns={...} />
  {!error && (
        <Table ... />}
</PageContainer>
```

This would make it so `TableLayout` only holds `tableInfo` and `modifyTableColumns`, the name is no longer great, but this makes it so that we are a bit more flexible.

Reply:
If it's only used in tables then I think the TableLayout is a good place and it's not necessary to move it into a separate component. If we feel like we want to move it out to a separate container when refactoring the patient page we can do that then. I will leave the TableLayout as it is now since I don't think that moving out the heading etc. presents more benefits.

Reply:
I'm gonna refactor CurrentSickleaves when this is merged so we can probably find even more improvements/changes then

Reply:
> If it's only used in tables then I think the TableLayout is a good place

It's not, that is why I suggested the change
![image](https://github.com/sklintyg/frontend/assets/28404/cb45c214-dc6b-412e-9092-007635e51d62)

This `div` could be considered general page layout, it would be inside Layout if it wasn't for Patient having a header that needs to be rendered above this element. Therefore, it's not really part of a Table construct and should be moved. This change in turn renders `TableLayout` a bit redundant except for holding the layout for `TableInfo` and `ModifyTableColumns`